### PR TITLE
fix(merge): merging RGB image with Grayscale should result in pixels resembling original images (MAPCO-4547)

### PR DESCRIPTION
**Pull request by shimoncohen**
| Question                | Answer                                                                          |
| ---------------- | -------------------------------------------------------------------------- |
| Bug fix         | ✔                                                                        |
| New feature     | ✖                                                                        |
| Breaking change | ✖                                                                        |
| Deprecations    | ✖                                                                        |
| Documentation   | ✖                                                                        |
| Tests added     | ✔                                                                        |
| Chore            | ✔                                                                       |

### CODE REVIEW

File: MergerLogicUnitTests/ImageProcessing/TileMergerTest.cs [hunk 1]
```diff
@@ -292,6 +334,8 @@ public void MergeTiles(Tile[] tiles, Coord targetCoord, TileFormatStrategy strat
             var tileBuilders = tiles.Select<Tile, CorrespondingTileBuilder>(tile => () => tile).ToList();
             var result = this._testTileMerger.MergeTiles(tileBuilders, targetCoord, strategy, uploadOnly);
 
+            File.WriteAllBytes("/home/shimonc1/Documents/GpkgMerger/MergerLogicUnitTests/ImageProcessing/TestImages/temp", result.GetImageBytes());
```
Comment (asafMasa) [inline MergerLogicUnitTests/ImageProcessing/TileMergerTest.cs]: please change this file location

File: MergerLogic/ImageProcessing/TileMerger.cs [hunk 2]
```diff
@@ -131,7 +131,11 @@ private void AddTileToImageList(Coord targetCoords, Tile? tile, List<MagickImage
             }
 
             var tileBytes = tile.GetImageBytes();
-            MagickImage? tileImage = new MagickImage(tileBytes);
+            MagickImage? tileImage = new MagickImage(tileBytes)
+            {
+                ColorSpace = ColorSpace.sRGB
```
Comment (asafMasa) [inline MergerLogic/ImageProcessing/TileMerger.cs:136]: this should be always?
don't we need to check the image type first?

**Review by asafMasa — CHANGES_REQUESTED** [review]

File: MergerLogicUnitTests/ImageProcessing/TileMergerTest.cs (hunk 1 above)
Comment (shimoncohen) [reply MergerLogicUnitTests/ImageProcessing/TileMergerTest.cs]: Removed, it was a mistake

File: MergerLogic/ImageProcessing/TileMerger.cs (hunk 2 above)
Comment (shimoncohen) [reply MergerLogic/ImageProcessing/TileMerger.cs:136]: Nope, it should always be `sRGB`, that's our target `ColorSpace` (see `MergeTiles` aswell).

**Review by asafMasa — APPROVED** [review]

**Review by ronenkapelian — APPROVED** [review]
good job honey

**Review by nitzanmorr — APPROVED** [review]
looks good to me

**Review by CL-SHLOMIKONCHA — APPROVED** [review]

**Pull request merged**